domain/ethereum.go

```diff
@@ -241,6 +363,66 @@ func (t Trace) ToProto() *protocol.TransactionEvent_Trace {
 	}
 }
 
+func TracesFromBlockData(bd *protocol.BlockData) ([]Trace, error) {
+	traces := make([]Trace, len(bd.Traces))
+	blockNumber, err := strconv.ParseInt(strings.Replace(bd.Block.Number, "0x", "", -1), 16, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	intBlockNumber := int(blockNumber)
+	for i, trace := range bd.Traces {
+		traces[i] = Trace{
+			BlockNumber: &intBlockNumber,
+			Subtraces:   int(trace.Subtraces),
+			TraceAddress: func() []int {
+				traceAddress := make([]int, len(trace.TraceAddress))
+				for i, address := range trace.TraceAddress {
+					traceAddress[i] = int(address)
+				}
+				return traceAddress
+			}(),
+			TransactionHash: &trace.TransactionHash,
+			TransactionPosition: func(i int64) *int {
```

Comment:
This looks confusing. Are we trying to use an anonymous function to convert `int64` to `*int`? I would just do that at the top, right after the left brace of the loop.

Reply:
we have a lot of cases of these pointer tricks. i think it's time to add a generic pointer-ref-getter method